doc/docusaurus/docs/essential-concepts/versions.md

```diff
@@ -0,0 +1,68 @@
+- At present, a newer ledger language version has access to more builtin functions than an older ledger language version.
+  Bear in mind that this will change as we plan to make all builtin functions available to all ledger language versions.
+- At present, Plutus Core version (explained below) 1.1.0 can only be used with Plutus V3, while Plutus V1 and V2 validators must use Plutus Core 1.0.0.
+  This will also change, as we plan to make all Plutus Core versions compatible with all ledger language versions.
+
+When do we introduce a new ledger language version?
+A new ledger language version must be introduced when a new ledger era is rolled out, because in a new ledger era, transactions will be different and contain new and/or modified fields, such as governance actions introduced in the Conway era.
+As a result, new information will need to be passed as arguments to Plutus validators.
+This necessitates a new ledger language version, because the arguments passed to existing ledger language versions must stay exactly the same, so that existing validators continue to work.
+
+Previously it was also the case that introducing new builtin functions or new Plutus Core versions necessitate a new ledger language version, but this is no longer the case.
+Introducing a new ledger language version is very expensive as it must be maintained forever.
+Going forward we expect the launch of new ledger eras to be the only reason for introducing new ledger language versions.
+
+## Plutus Core Version
+
+Plutus Core version is the usual sense of version pertaining to programming languages - in this instance the Plutus Core language.
+So far there have been two Plutus Core versions: 1.0.0 and 1.1.0.
+1.1.0 adds sums-of-products to the language by introducing two new AST node types: `Case` and `Constr`.
+See CIP-85 for more details.
+
+Note that adding new builtin functions does not require a new Plutus Core version.
+Once a new builtin function is added, one can simply start using the new builtin function with an existing Plutus Core version.
+
+## Builtin Semantics Variant
+
+Depending on the protocol version or the ledger language version, we may want to have different behavior of a particular builtin function.
+For example, we might want to fix a bug but need to retain the old buggy behavior for old evaluations, so we have to have a way of knowing which regime we are in.
+
+One example is the builtin function `consByteString`.
+If the first argument is outside of the range of `Word8`, the original behavior is to wrap it around.
+The new behavior starting at Plutus V3 is to throw an exception.
+
+When evaluating a standalone program using the `uplc` executable, flag `-S` or `--builtin-semantics-variant` can be used to inform the evaluator which semantics variant to use.
```

Comment:
This slightly contradicts the earlier:

> There is no need (and nowhere) to specify whether you are writing a Plutus V1, or Plutus V2, or Plutus V3 program

Since indirectly the ledger language version is linked to a builtin-semantics-variant, which is observable even when you run offline programs

Reply:
Suggestion to sth like: "omitting the flag defaults to latest/newest builtin semantics variant for each builtin"

Reply:
I don't see a contradiction. When evaluating a standlone program, you don't specify the ledger language version (since there's no ledger), but you can specify the builtin semantics variant. The fact that a particular ledger language version uses a particular builtin semantics variant doesn't contradict anything; it just happens to be that way. It's just a random correlation.

Reply:
> It's just a random correlation.

That's a weird thing to say. The builtin semantics variant is solely determined by the protocol version and the ledger language version, it's basically like a unique hash of the two. I'm against viewing it as some abstract concept that just happens to be there (and I'm against its very existence in the first place).